### PR TITLE
Throughput measurement

### PR DESCRIPTION
**Pull request by pkoryzna**

### CODE REVIEW

File: frontend/src/main/scala/akkaviz/frontend/FrontendApp.scala [hunk 1]
```diff
@@ -28,6 +27,8 @@ object FrontendApp extends JSApp with Persistence
 
   private val repo = new ActorRepository()
 
+  val MaxThroughputLogLen = 30
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/FrontendApp.scala]: Codacy Issue found: [Field name does not match the regular expression '^[a-z][A-Za-z0-9]*$'](https://www.codacy.com/app/lustefaniak/akka-viz/file/2145649411/issues/source?bid=3121282&fileBranchId=3151611#l30)

Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/FrontendApp.scala]: Codacy Issue found: [Field name 'MaxThroughputLogLen' does not match the regular expression '^[a-z][A-Za-z0-9]*$'](https://www.codacy.com/app/lustefaniak/akka-viz/file/2145649411/issues/source?bid=3121282&fileBranchId=3151611#l30)

File: frontend/src/main/scala/akkaviz/frontend/components/components.scala [hunk 2]
```diff
@@ -44,7 +46,29 @@ class ActorSelector(
     val content = div().render
     val stateVar = currentActorState(actor)
     stateVar.trigger {
+      import DOMGlobalScope.Rickshaw
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/components/components.scala]: Codacy Issue found: [Put imports in the beginning of the package](https://www.codacy.com/app/lustefaniak/akka-viz/file/2145649422/issues/source?bid=3121282&fileBranchId=3151611#l49)

File: monitoring/src/main/scala/akkaviz/events/GlobalSettingsActor.scala [hunk 3]
```diff
@@ -1,17 +1,23 @@
...
 
-/* Ugly singleton thing for things that cannot be done per-client,
-   but are not concerned with processing of the events.
+/* Ugly singleton thing for things that cannot (or shouldn't) be done per-client,
+   but are not concerned with publishing of the events.
   */
 class GlobalSettingsActor extends Actor with ActorLogging {
   private[this] var rcvDelay: FiniteDuration = 0.millis
   private[this] var eventPublisher: Option[ActorRef] = None
+  private[this] var throughputSrcRef: Option[ActorRef] = None
```
Comment (lustefaniak) [inline monitoring/src/main/scala/akkaviz/events/GlobalSettingsActor.scala]: Codacy Issue found: [Avoid mutable fields](https://www.codacy.com/app/lustefaniak/akka-viz/file/2145649434/issues/source?bid=3121282&fileBranchId=3151611#l18)

File: frontend/src/main/scala/akkaviz/frontend/FrontendApp.scala (hunk 1 above)
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/FrontendApp.scala]: Codacy Issue found: [Field name does not match the regular expression '^[a-z][A-Za-z0-9]*$'](https://www.codacy.com/app/lustefaniak/akka-viz/file/2145664781/issues/source?bid=3121282&fileBranchId=3151611#l30)

Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/FrontendApp.scala]: Codacy Issue found: [Field name 'MaxThroughputLogLen' does not match the regular expression '^[a-z][A-Za-z0-9]*$'](https://www.codacy.com/app/lustefaniak/akka-viz/file/2145664781/issues/source?bid=3121282&fileBranchId=3151611#l30)

File: frontend/src/main/scala/akkaviz/frontend/components/components.scala [hunk 4]
```diff
@@ -44,7 +45,27 @@ class ActorSelector(
...
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/components/components.scala]: Codacy Issue found: [Put imports in the beginning of the package](https://www.codacy.com/app/lustefaniak/akka-viz/file/2145664792/issues/source?bid=3121282&fileBranchId=3151611#l48)

File: monitoring/src/main/scala/akkaviz/events/GlobalSettingsActor.scala (hunk 3 above)
Comment (lustefaniak) [inline monitoring/src/main/scala/akkaviz/events/GlobalSettingsActor.scala]: Codacy Issue found: [Avoid mutable fields](https://www.codacy.com/app/lustefaniak/akka-viz/file/2145664804/issues/source?bid=3121282&fileBranchId=3151611#l18)

Comment (lustefaniak) [inline monitoring/src/main/scala/akkaviz/events/GlobalSettingsActor.scala]: Codacy Issue found: [Avoid mutable fields](https://www.codacy.com/app/lustefaniak/akka-viz/file/2145884983/issues/source?bid=3121282&fileBranchId=3151611#l18)

File: frontend/src/main/scala/akkaviz/frontend/FrontendApp.scala [hunk 5]
```diff
@@ -28,6 +28,8 @@ object FrontendApp extends JSApp with Persistence
...
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/FrontendApp.scala]: Codacy Issue found: [Field name does not match the regular expression '^[a-z][A-Za-z0-9]*$'](https://www.codacy.com/app/lustefaniak/akka-viz/file/2145884959/issues/source?bid=3121282&fileBranchId=3151611#l31)

Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/FrontendApp.scala]: Codacy Issue found: [Field name 'MaxThroughputLogLen' does not match the regular expression '^[a-z][A-Za-z0-9]*$'](https://www.codacy.com/app/lustefaniak/akka-viz/file/2145884959/issues/source?bid=3121282&fileBranchId=3151611#l31)

File: monitoring/src/main/scala/akkaviz/events/GlobalSettingsActor.scala (hunk 3 above)
Comment (lustefaniak) [inline monitoring/src/main/scala/akkaviz/events/GlobalSettingsActor.scala]: Codacy Issue found: [Avoid mutable fields](https://www.codacy.com/app/lustefaniak/akka-viz/file/2146051772/issues/source?bid=3121282&fileBranchId=3151611#l18)

File: frontend/src/main/scala/akkaviz/frontend/FrontendApp.scala (hunk 5 above)
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/FrontendApp.scala]: Codacy Issue found: [Field name does not match the regular expression '^[a-z][A-Za-z0-9]*$'](https://www.codacy.com/app/lustefaniak/akka-viz/file/2146051749/issues/source?bid=3121282&fileBranchId=3151611#l31)

Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/FrontendApp.scala]: Codacy Issue found: [Field name 'MaxThroughputLogLen' does not match the regular expression '^[a-z][A-Za-z0-9]*$'](https://www.codacy.com/app/lustefaniak/akka-viz/file/2146051749/issues/source?bid=3121282&fileBranchId=3151611#l31)

File: monitoring/src/main/scala/akkaviz/events/types/package.scala [hunk 6]
```diff
@@ -66,4 +66,6 @@ package object types {
 
   case class SnapshotAvailable(snapshot: LightSnapshot) extends BackendEvent
 
+  case class ThroughputMeasurement(actorRef: ActorRef, msgsPerSecond: Double, timestamp: Long = System.currentTimeMillis()) extends InternalEvent with BackendEvent
```
Comment (lustefaniak) [inline monitoring/src/main/scala/akkaviz/events/types/package.scala]: Codacy Issue found: [File line length exceeds 160 characters](https://www.codacy.com/app/lustefaniak/akka-viz/file/2150900131/issues/source?bid=3121282&fileBranchId=3151611#l69)

File: frontend/src/main/scala/akkaviz/frontend/DOMGlobalScope.scala [hunk 7]
```diff
@@ -15,4 +15,6 @@ object DOMGlobalScope extends js.GlobalScope {
 
   def str2ab(str: String): ArrayBuffer = js.native
 
+  def Rickshaw: js.Dynamic = js.native
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/DOMGlobalScope.scala]: Codacy Issue found: [Method name does not match the regular expression '^([a-z][A-Za-z0-9]*|<<|>>>|>|==|!=|<|<=|>|>=|\||&|\^|\+|-|\*|\/|%|\|\||&&|\+\+|--|\+=|-=)$'](https://www.codacy.com/app/lustefaniak/akka-viz/file/2150900103/issues/source?bid=3121282&fileBranchId=3151611#l18)

File: monitoring/src/main/scala/akkaviz/events/GlobalSettingsActor.scala (hunk 3 above)
Comment (lustefaniak) [inline monitoring/src/main/scala/akkaviz/events/GlobalSettingsActor.scala]: Codacy Issue found: [Avoid mutable fields](https://www.codacy.com/app/lustefaniak/akka-viz/file/2150900127/issues/source?bid=3121282&fileBranchId=3151611#l18)

**Comment by pkoryzna** [comment]
will re-work this into new ui

**Pull request closed without merging**